### PR TITLE
Port away from `window.Blockly.getMainWorkspace()`

### PR DESCRIPTION
**Pull request by Samq64**
### Changes

- Ports most addons to use `addon.tab.traps.getWorkspace()` instead of `Blockly.getMainWorkspace()`.
- Ports `reorder-custom-inputs` to use the `getMainWorkspace()` from `addon.tab.traps.getBlockly()`.
- The `Highlighter` class  used by the debugger and `editor-stepping` now takes an `addon` object but `setGlowingThreads()` could take the workspace directly if that's better.

### Reason for changes

After #7768 `window.Blockly` will no longer have `getMainWorkspace()` and it's a change that can be done now.

Some addons re-declare `Blockly` at their scope but I ported those too since `addon.tab.traps.getWorkspace()` is less prone to crashing the editor.

### Tests

Tested the basic functionality of each addon on Chromium on the main website. Most of these addons are still pretty broken on modern Blockly so I didn't test much there.

### CODE REVIEW

File: addons/find-bar/blockly/Utils.js [hunk 1]
```diff
@@ -18,11 +18,6 @@ export default class Utils {
     // this._myFlash = { block: null, timerID: null, colour: null };
     this.offsetX = 32;
     this.offsetY = 32;
-    this.navigationHistory = new NavigationHistory();
```
Comment (mxmou) [inline addons/find-bar/blockly/Utils.js:21]: I think it makes sense to keep `NavigationHistory` a separate class - it acts as a namespace for the methods related to navigation. You can add a constructor to it that takes `addon` as a parameter.

**Review by mxmou — APPROVED** [review]
Tested on Firefox.

**Comment by Samq64** [comment]
I can undo the ones that use `getBlockly()`'s version if you want to keep the changes to a minimum.

**Comment by WorldLanguages** [comment]
@Samq64 That might be a good idea, actually

In some edge cases, `blocklyObject.getMainWorkspace()` returns a different workspace (there's a few GitHub issues about that), and I'm not sure if it's a good idea to change this behavior unnecessarily. We might be fixing some bugs, creating new ones, or both, it's hard to know.

**Comment by WorldLanguages** [comment]
Related comment: https://github.com/ScratchAddons/ScratchAddons/pull/6227#issuecomment-1599952073

**Comment by WorldLanguages** [comment]
Will not merge this yet

**Comment by WorldLanguages** [comment]
I think this can be merged now.

**Pull request merged**